### DISCUSSION TITLE
Is it possible to use previous interface compiled wit with new bindings?

### DISCUSSION THREAD

**Discussion by canewsin** [wasmerio/wasmer#3111] [Q&A] [answered]
### Summary

I was exploring .wit format for plugin arch, every thing is ok until wit files had modification for next version. In general Plugins should be compatible with Prev version of wit bindings.

### Additional details
 V1
> plugin.wit
```rust
name: fn() -> string;
```

V2
> plugin.wit
```rust
name: fn() -> string;
description: fn() -> string;
```

Plugins compiled with V1 are not compatible with V2 imports, thus plugins needs recompiled for every version.

Is this possible with current wasmer wit compilers, with using different wit syntax ?

**Comment by epilys** [comment, marked as answer]
In the case of imports, these must be the same if you want the interface compatible with both versions.

For the export case, if everything else stays the same, V1 will just have extra exports that can be ignored.

**Comment by canewsin** [comment]
Ok, that might solve half the problem.
